lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -82,80 +160,100 @@ def is_active(self) -> bool:
             logger.error("Error while checking Vault health status: %s", e)
             return False
 
-    def is_api_available(self) -> bool:
-        """Return whether Vault is available."""
-        try:
-            self._client.sys.read_health_status(standby_ok=True)
-            return True
-        except (VaultError, RequestException) as e:
-            logger.error("Error while checking Vault health status: %s", e)
-            return False
-
-    def unseal(self, unseal_keys: List[str]) -> None:
-        """Unseal Vault."""
-        for unseal_key in unseal_keys:
-            self._client.sys.submit_unseal_key(unseal_key)
-        logger.info("Vault is unsealed")
-
-    def set_token(self, token: str) -> None:
-        """Set the Vault token for authentication."""
-        self._client.token = token
+    def enable_audit_device(self, device_type: AuditDeviceType, path: str) -> None:
+        """Enable a new audit device at the supplied path.
 
-    def approle_login(self, role_id: str, secret_id: str) -> None:
-        """Authenticate to Vault using AppRole."""
-        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+        Args:
+            device_type: One of three available device types
+            path: The path that will receive audit logs
+        """
+        try:
+            self._client.sys.enable_audit_device(
+                device_type=device_type.value,
+                options={"file_path": path},
+            )
+            logger.info("Enabled audit device %s for path %s", device_type.value, path)
+        except InvalidRequest as e:
+            errors = e.json.get("errors")
+            if len(errors) == 1 and errors[0].startswith("path is already in use"):
+                logger.info("Audit device already enabled.")
+            else:
+                raise e
+
+    def enable_approle_auth_method(self) -> None:
+        """Enable approle auth method."""
+        try:
+            self._client.sys.enable_auth_method("approle")
+            logger.info("Enabled approle auth method.")
+        except InvalidRequest as e:
+            errors = e.json.get("errors")
+            if len(errors) == 1 and errors[0].startswith("path is already in use"):
+                logger.info("Approle already enabled.")
+            else:
+                raise e
 
-    def configure_charm_access_policy(self, name: str) -> None:
-        """Create/update a policy within vault for the charm to access vault."""
-        with open("src/templates/charm_policy.hcl", "r") as f:
-            charm_policy = f.read()
-        self._client.sys.create_or_update_policy(name=name, policy=charm_policy)
-        logger.debug(f"Created charm policy: {name}")
+    def configure_policy(
```

Comment:
nit: Not a huge fan of these names (`configure_` when they're really creating/updating, but I realize that's a holdover. Feel free to ignore.